src/charm.py

```diff
@@ -182,6 +187,8 @@ def __init__(self, *args):
         observe(self.on.stop, self._on_stop)
         observe(self.on.update_status, self._on_update_status)
         observe(self.on.config_changed, self._on_config_changed)
+        observe(self.on.cert_transfer_relation_joined, self._on_cert_transfer_relation_joined)
+        observe(self.on.cert_transfer_relation_departed, self._on_cert_transfer_relation_departed)
```

Comment:
We need to call `self.cert_transfer.set_certificate` also on `self.cert.on.cert_changed`.

Reply:
good catch. fixed.